tests/sentry/monitors/consumers/test_monitor_consumer.py

```diff
@@ -687,6 +827,24 @@ def test_monitor_invalid_config(self):
             "my-invalid-monitor",
             monitor_config={"schedule": {"type": "crontab", "value": "13 * * * * *"}},
             environment="my-environment",
+            expected_error=CheckinValidationError(
+                [
+                    ProcessingError(
+                        ProcessingErrorType.MONITOR_INVALID_CONFIG,
+                        {
+                            "errors": {
+                                "schedule": [
+                                    ErrorDetail(
+                                        string="Only 5 field crontab syntax is supported",
+                                        code="invalid",
+                                    )
+                                ]
+                            }
+                        },
```

Comment:
The reason I kind of think dataclasses per type may have made sense is that we're going to have to have code somewhere that coerces this down to something consistent for the API

Reply:
This is just text returned from the validator, similar to what we'd return if they hit the api with the wrong info. We could define things more consistently if we need to, but I thought we'd just do something like
- Convert enum type to user friendly description
- Show blog of extra context to the user

Reply:
But if we want to render it better we can include more detail. Does having actual dataclasses help though? We basically just need to know which data will be present right?